woocommerce/woocommerce-ios
# Order Fulfillment: Allow the user to search and filter shipment providers

Implements #850 

This PR adds functionality to filter the list of shipment providers.

## Screenshots
All screenshots and recordings are on the iPhone 8 simulator
### Full List
![Simulator Screen Shot - iPhone 8 - 2019-04-12 at 14 00 32](https://user-images.githubusercontent.com/2722505/56015832-f01eb800-5d2c-11e9-9b18-ad620a804bba.png)

### Filtered list
![Simulator Screen Shot - iPhone 8 - 2019-04-12 at 14 00 36](https://user-images.githubusercontent.com/2722505/56015866-0af12c80-5d2d-11e9-8a6d-eb79b3ad5d8f.png)

### No results
![Simulator Screen Shot - iPhone 8 - 2019-04-12 at 14 10 21](https://user-images.githubusercontent.com/2722505/56015887-180e1b80-5d2d-11e9-921d-639b4eb67ea7.png)

### In action
![Screen Recording 2019-04-12 at 2 00 16 PM mov](https://user-images.githubusercontent.com/2722505/56015902-21978380-5d2d-11e9-9541-4161ad2df55f.gif)

## Diff
- Changed the way the `resultsController` is constructed in `ShippingProvidersViewModel`. I did not rename this class to `ShipmentProvidersViewModel` so that the diff could be isolated in the review. If / when this PR is ready to merge, we can rename this class.
- Renamed `ShippingProvidersViewController` to `ShipmentProvidersViewController`, added back the search bar, and pushed the result of users input into the view model so that the view model can filter the date
- Added some styles and setup the appearance of search bars
- Added some tests

## Testing
- Checkout the branch, run the tests
- Add a breakpoint in line 271 of `ShipmentProvidersViewController` (method `addCustomProvider`)
- Navigate to an order > Fulfill order > Add tracking > Shipping Provider
- Filter the list (i.e. typing DHL). Notice only providers matching the search term appear in the list
- Filter by a non-existent provider (for example `Zz`). That should present an "empty state" screen. 
- Tap the "Custom Provider" button, execution should stop at the breakpoint added previously.

Update release notes:
